### DISCUSSION TITLE
How to configure dependency that is hosted on github package

### DISCUSSION THREAD

**Discussion by stefan-g** [renovatebot/renovate#20954] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I got the following error :Renovate failed to look up the following dependencies: com.test:gradle-version-catalog.
Files affected: settings.gradle

I have a external dependency that i have published on github package (gpk). 

### What you tried so far.

I'm not quite sure how to configure that resource that the dependency can be looked up by renovate

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    }
  ],
  "hostRules": [
    {
      "matchHost": "https://maven.pkg.github.com/user/gradle-version-catalog",
      "hostType": "maven"
      "username": "user",
      "encrypted": {
        "token": "3f832f2983yf89hsd98ahadsjfasdfjaslf............"
      }
    }
  ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by stefan-g** [comment]
Tried to do it as described here: https://docs.mend.io/bundle/wsk/page/how_to_set_up_hostrules_for_gradle_private_registries_hosted_in_github_packages.html

but encrypted token does not seems to work i got the following excpetion:
   "message": "Error decrypting message: Session key decryption failed."

How to set a token? is event required as renovate is running on github as well?

**Comment by secustor** [comment]
You have to look at the other sections of https://docs.mend.io/bundle/wsk/page/how_to_set_up_hostrules_for_gradle_private_registries_hosted_in_github_packages.html

The hostRule has to contain the encrypted value and your host seems to be wrong: 
```
{
  "hostRules": [
    {
      "matchHost": "https://maven.pkg.github.com/OWNER/REPOSITORY",
      "hostType": "gradle",
      "userName": "hostrules",
      "encrypted": {
        "token": "3f832f2983yf89hsd98ahadsjfasdfjaslf............"
      }
    }
  ]
}
```

**Comment by stefan-g** [comment]
Yes i tried it also with encrypted token but got the following error:

DEBUG: Trying to decrypt token
DEBUG: Could not decrypt using openpgp
{
  "err": {
    "message": "Error decrypting message: Session key decryption failed.",
    "stack": "Error: Error decrypting message: Session key decryption failed.\n    at Eh.decryptSessionKeys (/home/ubuntu/renovateapp/node_modules/openpgp/dist/node/openpgp.min.js:2:338796)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Eh.decrypt (/home/ubuntu/renovateapp/node_modules/openpgp/dist/node/openpgp.min.js:2:335902)\n    at async Object.exports.decrypt (/home/ubuntu/renovateapp/node_modules/openpgp/dist/node/openpgp.min.js:16:132527)\n    at async tryDecryptPgp (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/decrypt.js:36:26)\n    at async tryDecrypt (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/decrypt.js:82:33)\n    at async decryptConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/decrypt.js:157:40)\n    at async decryptConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/decrypt.js:204:47)\n    at async mergeRenovateConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/merge.js:212:29)\n    at async getRepoConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:11:14)\n    at async initRepo (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/index.js:34:14)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:45:18)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:260:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:454:5"
  }
}
DEBUG: Trying to decrypt with old private key
DEBUG: Could not decrypt using default padding
DEBUG: Could not decrypt using PKCS1 padding

 Is the documentation still up to date? how do i create a encrypted token?
